doc/terminology.md

```diff
@@ -0,0 +1,86 @@
+
+
+ ┌────────────────┐
+ │                │
+ │  Provider      │
+ │                │
+ │                │
+ └────────────────┘
+```
+
+Usually, when using KUKSA.val you will run one instance of KUKSA.val databroker. Databroker can be seen as the "core" of a KUKSA.val deployment: It manages the VSS signals and exposes them via an API. Databroker is also responsible of for checking whether a client has sufficient authorisation to access certain signals.
+
+Databroker's API is GRPC-based, which means it can be accessed over the network by other programs or systems. From an architecture point of view, there are different components, with different roles interacting with databroker explained in the following. While a given system or binary might combine several of these roles, it is helpful to think in these categorizations when desgning a system.
+
+For more specific examples of components fullfilling these roles also check [System Architecture](./system-architecture.md).
+**todo: probably system architecture not a good name for that file, and I did not update it with the terms in this doc**
+
+
+### Clients
+**Discussion: App (may be to narrow)? User (i thinkw as called that in wiki. but clashes then with "end-user" or developer using KUKSA.val? On the other hand client may also be a good "overall" term for everything contacting our API**
+
+A client is a program that intends to consume the VSS data points. This includes reading of sensors or actuating actors. A client intends to use capabilites offered by a vehicle.
+
+So typically this might be "apps" running on vehicle computers, the infotainment, or maybe even a user's personal device. 
+
+### Providers
+A provider is a program that makes sure interactions with the VSS data points are linked to the state of a vehicle.
+
+A provider can can supply either be a
+ *  *sensor-provider*, providing information for a sensor defined in VSS, i.e. a component that updates `Vehicle.Speed` based on the actual speed of an vehicle.
+ * An *actuator-provider*, trying to act on the target value of an actuator defined in VSS, i.e. a component that based on the target value of `Vehicle.Body.Trunk.Rear.IsOpen` tries to ensure that the actual state of the trunk (open or closed) matches the desired state set by clients.
+
+ ## Vehicle Signal Specification (VSS)
+ KUKSA.val handles datapoints that are defined using the COVESA Vehicle Specification (VSS).
+
+ VSS introduces a domain taxonomy for vehicle signals. It defines
+ * A syntax for defining vehicle signals in a structured manner.
+ * A catalog of standard signals (VSs standard catalogue) related to vehicles.
+
+It can be used as standard in automotive applications to communicate information around the vehicle, which is semantically well defined. 
+
+See the [the official VSS documentation](https://covesa.github.io/vehicle_signal_specification/) for all details. Here we will summarizes the terms and concepts most relevant for KUKSA.val.
```

Comment:
The `User` / `Provider` roles can also be seen as analogous to the more common `User` / `System` roles.

Providers are the system services.
Users are the, well, users.

Reply:
In this revison I have chosen “Consumer” instead of (VSS) User. I think replacing every mention of “Consumer” with User” works as well. Reasons for going with consumer
 - It feels a nice counterpart for “provide”
 - No danger of clashes with human endusers
Potential reasons against
 - One might not feel comfortable with a “consumer” doing actuation. (I think it is ok, becasue as “consumer” of my power drill I am also free to make holes.)

But maybe somebody there is a term I haven’t thought about

I tried not to get too deeply technical with mentioning “subscribe” or“set”.  My thinking model was rather “digital twin” (without mentioning that particularly loaded word :) ) whereas my goal is to learn something  or changing something  about the true state via interacting with VSS. Thus, the role of Providers basically becomes syncing state.

Tell me if it worked.

Reply:
> I think it is ok, becasue as “consumer” of my power drill I am also free to make holes.

Haha :)
Yes, I think consumer gets the point across as well.